### PR TITLE
movement snake

### PR DESCRIPTION
**Pull request by JacksonG8s**
Added notes, tick based movement, Eventlistener for user input, user input states, and update intervals.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -18,15 +18,27 @@
...
     const ctx = canvas.getContext("2d");
     var globalID;
     var score;
+    var playerMovement;
+    var userInput;
     var gameState = Array.from(Array(40), () => new Array(40).fill(0));
+    var keyW = false;
+    var keyA = false;
+    var keyS = false;
+    var keyD = false;
```
Comment (Tyler-Gates) [inline index.html:32]: can simplify this down to a string based state storage, something like `var snakeDirection = "right"` that could equal 'left', 'up', 'right', 'down'

File: index.html [hunk 2]
```diff
@@ -74,10 +90,70 @@
...
     }
-    function wiggles() {}
-    function score() {
-      for (var b = 0; b < 40; b++) {
-        for (var i = 0; i < 40; i++) {}
+    //Movement Input Register
+    window.addEventListener("keydown", (event) => wiggleKeyDown());
+    function wiggleKeyDown() {
+      var keyCode = event.keyCode;
+      console.log(keyCode);
```
Comment (Tyler-Gates) [inline index.html:97]: remove console.log here

**Review by Tyler-Gates — CHANGES_REQUESTED** [review]

File: index.html [hunk 3]
```diff
@@ -18,15 +18,27 @@
     </div>
   </body>
   <script>
+    //States
     const canvas = document.getElementById("snake");
     const ctx = canvas.getContext("2d");
     var globalID;
     var score;
+    var playerMovement;
+    var userInput;
```
Comment (Tyler-Gates) [inline index.html:27]: these are not unused variables

File: index.html [hunk 4]
```diff
@@ -18,15 +18,27 @@
...
+    var x = 0;
+    var y = 0;
+
+    //Initialization of snake and apple
     gameState[20][20] = 2;
     gameState[0][0] = 1;
 
-    function updateGameStateMap() {
+    //Checks position of matrix, and updates color of 10x10 pixel square
+    function drawGameStateMap() {
```
Comment (Tyler-Gates) [inline index.html:41]: I like the name here! Very descriptive of it's function

File: index.html (hunk 3 above)
Comment (JacksonG8s) [reply index.html:27]: Removed excess variables.

File: index.html (hunk 2 above)
Comment (JacksonG8s) [reply index.html:97]: Removed console log.

File: index.html (hunk 1 above)
Comment (JacksonG8s) [reply index.html:32]: Resolved and added inputs for arrow keys.

**Review by Tyler-Gates — APPROVED** [review]
LGTM

**Pull request merged**